app/javascript/alchemy_admin/dialog.js

```diff
@@ -0,0 +1,127 @@
+import { createHtmlElement } from "alchemy_admin/utils/dom_helpers"
+
+export class Dialog {
+  #dialogComponent
+  #onReject
+  #onResolve
+
+  /**
+   * @param {string} url
+   * @param {object} options
+   */
+  constructor(url, options = {}) {
+    this.url = url
+    this.options = { title: "", size: "300x400", padding: true, ...options }
+  }
+
+  /**
+   * load the content of given url and than open the dialog
+   * @returns {Promise<unknown>}
+   */
+  open() {
+    this.#build()
+    // Show the dialog with the spinner after a small delay.
+    // Most of the times the spinner will not be visible, because the requested
+    // content is already downloaded. These behavior "feels" a little bit better.
+    // It calls the show - method twice, but this is not a big deal.
+    setTimeout(() => this.#dialogComponent.show(), 300)
```

Comment:
this should not be necessary if we show the dialog before we load the remote content